File: apps/apple-tv/MediaPlayer/Models/Playlist.swift
import Foundation

struct Playlist: Identifiable, Codable, Hashable {
    let id: Int64
    let userId: Int64
    let name: String
    let description: String?
    let itemCount: Int
    let createdAt: String
    let updatedAt: String

    enum CodingKeys: String, CodingKey {
        case id, name, description
        case userId = "user_id"
        case itemCount = "item_count"
        case createdAt = "created_at"
        case updatedAt = "updated_at"
    }
}

struct PlaylistItem: Identifiable, Codable, Hashable {
    let id: Int64
    let playlistId: Int64
    let mediaId: Int64
    let mediaType: MediaType
    let position: Int
    let addedAt: String
    let title: String
    let year: Int?
    let posterPath: String?
    let duration: Int?
    let overview: String?
    let rating: Double?
    let resolution: String?

    enum CodingKeys: String, CodingKey {
        case id, position, title, year, duration, overview, rating, resolution
        case playlistId = "playlist_id"
        case mediaId = "media_id"
        case mediaType = "media_type"
        case addedAt = "added_at"
        case posterPath = "poster_path"
    }

    var formattedDuration: String {
        guard let duration = duration else { return "" }
        let hours = duration / 3600
        let minutes = (duration % 3600) / 60
        if hours > 0 {
            return "\(hours)h \(minutes)m"
        }
        return "\(minutes)m"
    }
}

struct PlaylistWithItems: Codable {
    let playlist: Playlist
    let items: [PlaylistItem]
}

struct CreatePlaylistRequest: Codable {
    let name: String
    let description: String?
}

struct ReorderPlaylistRequest: Codable {
    let itemIds: [Int64]

    enum CodingKeys: String, CodingKey {
        case itemIds = "item_ids"
    }
}

struct PlaylistsResponse: Codable {
    let items: [Playlist]
}

File: apps/fire-tv/app/src/main/java/com/mediaserver/tv/data/models/Playlist.kt
package com.mediaserver.tv.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Playlist(
    val id: Long,
    @SerialName("user_id") val userId: Long,
    val name: String,
    val description: String? = null,
    @SerialName("item_count") val itemCount: Int = 0,
    @SerialName("created_at") val createdAt: String,
    @SerialName("updated_at") val updatedAt: String
)

@Serializable
data class PlaylistItem(
    val id: Long,
    @SerialName("playlist_id") val playlistId: Long,
    @SerialName("media_id") val mediaId: Long,
    @SerialName("media_type") val mediaType: MediaType,
    val position: Int,
    @SerialName("added_at") val addedAt: String,
    val title: String,
    val year: Int? = null,
    @SerialName("poster_path") val posterPath: String? = null,
    val duration: Int? = null,
    val overview: String? = null,
    val rating: Double? = null,
    val resolution: String? = null
) {
    val formattedDuration: String
        get() {
            val d = duration ?: return ""
            val hours = d / 3600
            val minutes = (d % 3600) / 60
            return if (hours > 0) "${hours}h ${minutes}m" else "${minutes}m"
        }

    val posterUrl: String?
        get() = posterPath?.let { "https://image.tmdb.org/t/p/w185$it" }
}

@Serializable
data class PlaylistsResponse(
    val items: List<Playlist>
)

@Serializable
data class PlaylistWithItems(
    val playlist: Playlist,
    val items: List<PlaylistItem>
)

@Serializable
data class CreatePlaylistRequest(
    val name: String,
    val description: String? = null
)

@Serializable
data class ReorderPlaylistRequest(
    @SerialName("item_ids") val itemIds: List<Long>
)
